### PR TITLE
[Fix-16578] Fix environment resource permission

### LINKED ISSUE #16578: [Bug] [api] Only admin users can operate the environment management
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

Only admin users can view the environment management page, ordinary users cannot see it, but they can call the interface.

### What you expected to happen

Only admin users can operate the environment management page

### How to reproduce

Ordinary users call the interface with their own login information
/dolphinscheduler/environment/update Can be called successfully

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wangxj3**
## Purpose of the pull request

fix #16578

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

## Pull Request Notice
[Pull Request Notice](https://github.com/apache/dolphinscheduler/blob/dev/docs/docs/en/contribute/join/pull-request.md)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckServiceImpl.java [hunk 1]
```diff
@@ -267,6 +270,10 @@ public List<AuthorizationType> authorizationTypes() {
 
         @Override
         public boolean permissionCheck(int userId, String url, Logger logger) {
+            User user = processService.getUserById(userId);
+            if (user.getUserType() != UserType.ADMIN_USER) {
+                return false;
+            }
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckServiceImpl.java]: This is too restrictive. We just need to add environment control.

**Review by SbloodyS — CHANGES_REQUESTED** [review]

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckServiceImpl.java (hunk 1 above)
Comment (wangxj3) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckServiceImpl.java]: > This is too restrictive. We just need to add environment control.

Do you want to add this feature to the service?

Comment (SbloodyS) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckServiceImpl.java]: Only environment permissions need to be controlled.

Comment (wangxj3) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckServiceImpl.java]: 图片

This method only use in environment permissions

**Review by SbloodyS — DISMISSED** [review]
LGTM

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckServiceImpl.java [hunk 2]
```diff
@@ -256,6 +256,9 @@ public static class EnvironmentResourcePermissionCheck implements ResourceAcquis
 
         private final EnvironmentMapper environmentMapper;
 
+        @Autowired
+        private ProcessService processService;
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckServiceImpl.java]:
```suggestion
        private UserDao userDao;
```

Comment (wangxj3) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckServiceImpl.java]: ok

**Review by ruanwenjun — DISMISSED** [review]
LGTM

**Review by SbloodyS — DISMISSED** [review]

**Comment by SbloodyS** [comment]
You should check failed UT. @wangxj3

**Comment by wangxj3** [comment]
> You should check failed UT. @wangxj3

ok

**Review by SbloodyS — APPROVED** [review]

**Review by ruanwenjun — APPROVED** [review]

**Pull request merged**